lib/metadata/util/win32/Win32Software.rb

```diff
@@ -251,11 +251,13 @@ def convert_times(attrs)
     end
 
     def self.DecodeProductKey(product_key)
-      return if product_key.blank? || product_key.length < 67
-      y = []; product_key.split(",")[52..67].each { |b| y << b.hex }
-      return MIQEncode.base24Decode(y)
+      parts = product_key.to_s.split(",")
+      return if parts.length < 67
```

Comment:
the code before was checking `product_key.length`, whereas the current code is checking `parts.length`.  Why the switch?

Reply:
I had put it in the commit message, but forgot to put it in the OP

> The length check was incorrect as it checked the length of the string
> and not the length of the Array after splitting.